Conductor/Client/ApiClient.cs

```diff
@@ -160,20 +161,49 @@ private byte[] GetBytes(Stream input)
         /// <param name="contentType">Content Type of the request</param>
         /// <returns>Object</returns>
         public Object CallApi(
-            String path, RestSharp.Method method, List<KeyValuePair<String, String>> queryParams, Object postBody,
+            String path, Method method, List<KeyValuePair<String, String>> queryParams, Object postBody,
             Dictionary<String, String> headerParams, Dictionary<String, String> formParams,
             Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams,
-            String contentType)
+            String contentType, Configuration configuration = null)
         {
-            var request = PrepareRequest(
-                path, method, queryParams, postBody, headerParams, formParams, fileParams,
-                pathParams, contentType);
+            RestResponse response = null;
+            bool isTokenRefreshed = false;
+            do
+            {
+                var request = PrepareRequest(
+                    path, method, queryParams, postBody, headerParams, formParams, fileParams,
+                    pathParams, contentType);
+
+                response = ExecuteRestClientAPI(request, method);
 
+                if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
+                {
+                    var JsonContent = JsonConvert.DeserializeObject<JObject>(response.Content);
+
+                    if (JsonContent["error"].ToString() == "EXPIRED_TOKEN" && !isTokenRefreshed)
+                    {
+                        string refreshToken = configuration.GetRefreshToken();
+                        headerParams["X-Authorization"] = refreshToken;
+                        isTokenRefreshed = true;
+                    }
+                    else
+                        isTokenRefreshed = false;
+                }
+                else
+                    isTokenRefreshed = false;
+            } while (isTokenRefreshed);
```

Comment:
1. Why do we have to refresh the token in an infinite loop here? Given server can/ can not be reachable?
2. Please write tests for the scenario covering 4xx response and with refreshToken() things will work.